### ISSUE TITLE
Altered bubble sort fails: 'Subscript' object has no attribute 'id'

### ISSUE THREAD

**Issue by jayvdb** [Coronon/PySchemeTranspiler#12]
The original python is internally rewritten by py2many to be https://github.com/adsharma/py2many/blob/main/tests/expected/bubble_sort.py
```py
def bubble_sort(seq: List[int]) ->List[int]:
    l = len(seq)
    for _ in range(l):
        for n in range(1, l):
            if seq[n] < seq[n - 1]:
                if True:
                    __tmp1, __tmp2 = seq[n], seq[n - 1]
                    seq[n - 1]: int = __tmp1
                    seq[n]: int = __tmp2
    return seq


if True:
    unsorted: List[int] = [14, 11, 19, 5, 16, 10, 19, 12, 5, 12]
    expected: List[int] = [5, 5, 10, 11, 12, 12, 14, 16, 19, 19]
    assert bubble_sort(unsorted) == expected
    print('OK')
```

py2rb chokes on the `seq[..] = ..`.  I can disable that rewriter.

But then it also chokes on `unsorted: List[int] = [14, 11, 19, 5, 16, 10, 19, 12, 5, 12]` producing error
> [TypeError] Can not redefine variable type 

And I really dont want to disable that rewriter.

**Comment by Coronon** [comment]
The original bubble_sort.py works perfectly well in the current version. You do not have to create those __temp variables. (Tested without the `if True:`)

**Comment by Coronon** [comment]
The second error I think is due to the `if True:` putting the define before the if-scope. I will investigate (Tomorrow, as I don’t have time today).

**Comment by Coronon** [comment]
Update: Sitting in a train, but already found the issue. Its in fact that AnnAssign blocks this as a redefinition in strict-mode. The fix will be a bit more complex in the typing system, so expect one tomorrow after work ;)

**Comment by Coronon** [comment]
That the transpiler cant handle the first part is actually completly correct as its invalid Python code:
![image](https://user-images.githubusercontent.com/33808743/124761467-1f372a80-df32-11eb-9d27-ab8edf447852.png)

And again: You do not need those `__temp` variables anymore, the online switch works just as fine (if not as efficient I have to admit) :)

**Comment by Coronon** [comment]
The assignment bug is fixed in: 5bc342206b3e157c74bd0da4c129a643565baa8c
